lib/toolbox/keys_dict.c

```diff
@@ -0,0 +1,324 @@
+#include "keys_dict.h"
+
+#include <storage/storage.h>
+#include <flipper_format/flipper_format.h>
+#include <toolbox/stream/file_stream.h>
+#include <toolbox/stream/buffered_file_stream.h>
+#include <toolbox/args.h>
+
+#define TAG "KeysDict"
+
+struct KeysDict {
+    Stream* stream;
+    size_t key_size;
+    size_t key_size_symbols;
+    size_t total_keys;
+};
+
+static inline void __add_ending_new_line(KeysDict* instance) {
+    if(stream_seek(instance->stream, -1, StreamOffsetFromEnd)) {
+        uint8_t last_char = 0;
+
+        // Check if the last char is new line or add a new line
+        if(stream_read(instance->stream, &last_char, 1) == 1 && last_char != '\n') {
+            FURI_LOG_D(TAG, "Adding new line ending");
+            stream_write_char(instance->stream, '\n');
+        }
+
+        stream_rewind(instance->stream);
+    }
+}
+
+static inline bool __read_key_line(KeysDict* instance, FuriString* line, bool* is_endfile) {
+    if(stream_read_line(instance->stream, line) == false)
+        *is_endfile = true;
+
+    else {
```

Comment:
Please, add here and below brackets {}. It will be much easier to read and allows to avoid mistakes in future.